### PR TITLE
feat: get new block signals from websocket

### PR DESCRIPTION
**Pull request by Ivan-Feofanov**

### CODE REVIEW

**Review by dputko — COMMENTED** [review]
excuse me,  i am долблюсь в глаза,  but where is some websocket stuff?

**Comment by Ivan-Feofanov** [comment]
here: https://github.com/sensority-labs/big-ear/pull/2/files#diff-bdbd92f986186abb28c68164ecc3196750793fe14aa96e289d864581d6621eddR39

File: pkg/config/config.go [hunk 1]
```diff
@@ -29,7 +30,7 @@ type Config struct {
 	Providers   Providers
 	Streams     Streams
 	LastDataDir string `default:"/tmp/"`
-	Block       uint64
+	Block       int64
```
Comment (dputko) [inline pkg/config/config.go:33]: Can the block number be negative?

File: pkg/network/network.go [hunk 2]
```diff
@@ -1 +1,78 @@
...
+}
+
+func (c *WSClient) ListenNewBlocks(ch chan<- json.RawMessage, errCh chan<- error) {
+	u, err := url.Parse(c.url)
+	if err != nil {
+		errCh <- errtrace.Wrap(err)
+	}
+
+	log.Default().Println("Connecting to", u.Host)
+	ws, err := websocket.Dial(c.url, "", "http://localhost")
```
Comment (dputko) [inline pkg/network/network.go:46]: Don't you think it's better to make "localhost" env variable?

File: pkg/network/network.go [hunk 3]
```diff
@@ -1 +1,78 @@
...
+	if err != nil {
+		errCh <- errtrace.Wrap(err)
+	}
+	defer func(c *websocket.Conn) {
+		err := c.Close()
+		if err != nil {
+			errCh <- errtrace.Wrap(err)
```
Comment (dputko) [inline pkg/network/network.go:53]: errtrace is the "Go way" to achive stack traces, is't it?

File: pkg/network/network.go [hunk 4]
```diff
@@ -1 +1,78 @@
...
+		errCh <- errtrace.Wrap(err)
+	}
+
+	// read a subscription message
+	var subscriptionID []byte
+	if err := websocket.Message.Receive(ws, &subscriptionID); err != nil {
+		errCh <- errtrace.Wrap(err)
+	}
+
+	for {
```
Comment (dputko) [inline pkg/network/network.go:67]: How are we supposed to stop listening and unsubscribe?

**Pull request merged**